calibration/test/e2e_test.jl

```diff
@@ -0,0 +1,169 @@
+        simdir_path = joinpath(member_path, "output_active")
+        if isdir(simdir_path)
+            simdir = SimDir(simdir_path)
+            G_ensemble[:, m] .= process_member_data(simdir)
+        else
+            G_ensemble[:, m] .= NaN
+        end
+    end
+    return G_ensemble
+end
+
+function process_member_data(simdir::SimDir)
+    isempty(simdir.vars) && return NaN
+    rsut =
+        get(simdir; short_name = "rsut", reduction = "average", period = "30d")
+    return slice(average_xy(rsut); time = 30).data
+end
+
+# Generate observations
+obs_path = joinpath(experiment_dir, "observations.jld2")
+if !isfile(obs_path)
+    @info "Generating observations"
+    config = CA.AtmosConfig(joinpath(experiment_dir, "model_config.yml"))
+    simulation = CA.get_simulation(config)
+    CA.solve_atmos!(simulation)
+    observations = Vector{Float64}(undef, 1)
+    observations .= process_member_data(SimDir(simulation.output_dir))
+    JLD2.save_object(obs_path, observations)
+end
+
+# Initialize experiment data
+astronomical_unit = 149_597_870_000
+observations = JLD2.load_object(obs_path)
+noise = 0.1 * I
+n_iterations = 3
+ensemble_size = 10
+prior = CAL.get_prior(joinpath(experiment_dir, "prior.toml"))
+output_dir = "calibration_end_to_end_test"
+experiment_config = CAL.ExperimentConfig(;
+    n_iterations,
+    ensemble_size,
+    observations,
+    noise,
+    output_dir,
+    prior,
+)
+
+# Minimal EKI test
+function minimal_eki_test(julia_eki)
+    params = EKP.get_u(julia_eki)
+    spread = map(x -> var(abs.(x)), params)
+
+    # Spread should be heavily decreased as particles have converged
+    @test last(spread) / first(spread) < 0.1
+    # Parameter should be close to true value
+    @test mean(last(params)) ≈ astronomical_unit rtol = 0.02
+end
+
+# Caltech HPC backend
+backend = CAL.get_backend()
+@info "Running Calibration E2E test on $backend"
+if backend <: CAL.SlurmBackend
+    slurm_kwargs = CAL.kwargs(time = 5)
+    test_eki = CAL.calibrate(
+        backend,
+        experiment_config;
+        slurm_kwargs,
+        model_interface,
+        verbose = true,
+    )
+else
+    test_eki = CAL.calibrate(backend, experiment_config)
+end
+@testset "Test Calibration on $backend" begin
+    minimal_eki_test(test_eki)
+end
+
+# Run calibration
+julia_eki = CAL.calibrate(CAL.JuliaBackend, experiment_config)
+minimal_eki_test(julia_eki)
+
+@testset "Julia-only comparison calibration" begin
+    minimal_eki_test(julia_eki)
+end
+
+@testset "Compare $backend output to JuliaBackend" begin
+    for (uu, slurm_uu) in zip(EKP.get_u(julia_eki), EKP.get_u(test_eki))
+        @test uu ≈ slurm_uu rtol = 0.02
+    end
+end
+
+# Debug plots
+function scatter_plot(julia_eki::EKP.EnsembleKalmanProcess)
+    f = CairoMakie.Figure(resolution = (800, 600))
+    ax = CairoMakie.Axis(
+        f[1, 1],
+        ylabel = "Absolute Parameter Value",
+        xlabel = "TOA Radiative SW Flux 30day average",
```

Comment:
I think that this label is incorrect

Reply:
Could you clarify this? It looks correct to me

Reply:
Yes, you are right, I thought this was something else.